src/dmd/lambdacomp.d

```diff
@@ -59,8 +59,26 @@ private enum ExpType
  * variable or a template instance is encountered, the
  * serialization is dropped and the function is considered
  * uncomparable.
+ *
+ * Params:
+ *  fld = the starting AST node for the lambda function
+ *  sc = the scope in which the lambda function is located
+ *
+ * Returns:
+ *  The serielization of `fld`.
  */
-extern (C++) class SerializeVisitor : SemanticTimeTransitiveVisitor
+string getSerialization(FuncLiteralDeclaration fld, Scope* sc)
+{
+    auto serVisitor = new SerializeVisitor(sc);
+    fld.accept(serVisitor);
+    auto len = serVisitor.buf.offset;
+    if (len == 0)
+        return null;
+    else
+        return serVisitor.buf.peekString()[0 .. len].idup;
```

Comment:
`return serVisitor.buf.extractString()[0 .. len];`
The extract takes ownership of the buffer, so it does not need to be duplicated.